src/components/Admin/UserEdit.tsx

```diff
@@ -0,0 +1,138 @@
+import { gql } from '@apollo/client';
+import generator from 'generate-password';
+import React, { FC, useState } from 'react';
+import { Alert, FormGroup } from 'react-bootstrap';
+import { Prompt, useHistory } from 'react-router-dom';
+import { EditMode, UserForm } from '.';
+import { useCreateUserMutation, UserInput, useUpdateUserMutation } from '../../generated/graphql';
+import { USER_DETAILS_FRAGMENT } from '../../graphql/admin';
+import { UserModel } from '../../models/User';
+import { Loading } from '../core/Loading';
+import InputWithCopy from './InputWithCopy';
+import User from './User';
+
+interface UserEditProps {
+  editMode?: EditMode;
+  title?: string;
+}
+
+export const UserEdit: FC<UserEditProps> = ({ editMode = EditMode.readOnly, title = 'User' }) => {
+  const [showSuccess, setShowSuccess] = useState<boolean>(false);
+  const [showError, setShowError] = useState<boolean>(false);
+  const [strongPassword, setStrongPassword] = useState<string>('');
+  const [isBlocked, setIsBlocked] = useState<boolean>(false);
+  const history = useHistory();
+  const [updateUser, { loading: updateMutationLoading }] = useUpdateUserMutation({
+    onError: error => console.log(error),
+    onCompleted: () => setShowSuccess(true),
+  });
+
+  const isEditMode = editMode === EditMode.edit;
+
+  const [createUser, { loading: createMutationLoading }] = useCreateUserMutation({
+    onError: error => {
+      setShowError(true);
+      console.log(error);
+    },
+    onCompleted: data => {
+      history.replace(`/admin/users/${data.createUser.id}/edit`);
+      setIsBlocked(true);
+      setShowSuccess(true);
```

Comment:
add setShowError(false) in case previous request failed but current passed

same for onError. add setShowSuccess(false)